### DISCUSSION TITLE
Support for async JS functions

### DISCUSSION THREAD

**Discussion by bmcbride** [Budibase/budibase#3324] [Help]
Hello! Are async JavaScript functions supported in button actions or embeds? I don't believe so, but wanted to see if anyone had any solutions or workarounds. Thanks!

**Comment by mjashanks** [comment]
Nope not at the moment. You might be able to get them to work with Embeds, depending on what you actually need to do. The issue with Embed is that you can get data into it, but cannot get data back out / read from it. So, your embedded function can't return anything.
